provider/src/test/kotlin/org/gradle/kotlin/dsl/NamedDomainObjectCollectionExtensionsTest.kt

```diff
@@ -110,6 +112,21 @@ class NamedDomainObjectCollectionExtensionsTest {
         }
     }
 
+    @Test
+    fun `can access existing element by getting with reified type`() {
+        val element = DomainObject()
+        val container = mock<PolymorphicDomainObjectContainer<SomeObject>> {
+            on { getByName("domainObject") } doReturn element
+        }
+
+        container { // invoke syntax reified method
+            val domainObject by getting<DomainObject>()
```

Comment:
:o: is `val domainObject: DomainObject by getting` also allowed?

Reply:
Yes, there is a test above that covers that.